modules/accredited_representative_portal/app/services/accredited_representative_portal/saved_claim_service/attach.rb

```
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module AccreditedRepresentativePortal
+  module SavedClaimService
+    module Attach
+      Error = Class.new(RuntimeError)
+      InvalidFileError = Class.new(Error)
+
+      class << self
+        def perform(file, is_form:)
+          klass =
+            if is_form
+              PersistentAttachments::VAForm
+            else
+              PersistentAttachments::VAFormAttachment
+            end
+
+          klass.new.tap do |attachment|
+            attachment.file = file
+            validate_record!(attachment)
+            validate_upstream!(attachment)
+
+            attachment.save
+          end
+
+        ##
+        # Expose a discrete set of known exceptions. Expose any remaining with a
+        # catch-all exception.
+        #
+        rescue InvalidFileError
+          raise
+        rescue
+          raise Error
```

I don't understand why we are masking all other errors with a generic one here? Won't this make troubleshooting harder?

If we need to show a generic message to the user, vets-website can handle that.